ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java

```diff
@@ -159,57 +173,12 @@ void handleConnect(final EthPeer peer) {
   public ValidationResult<TransactionInvalidReason> addTransactionViaApi(
       final Transaction transaction) {
 
-    if (configuration.getDisableLocalTransactions()) {
-      final var result = addRemoteTransaction(transaction);
-      if (result.isValid()) {
-        transactionBroadcaster.onTransactionsAdded(List.of(transaction));
-      }
-      return result;
-    }
-    return addLocalTransaction(transaction);
-  }
-
-  ValidationResult<TransactionInvalidReason> addLocalTransaction(final Transaction transaction) {
-    final ValidationResultAndAccount validationResult = validateLocalTransaction(transaction);
-
-    if (validationResult.result.isValid()) {
-
-      final TransactionAddedResult transactionAddedResult =
-          pendingTransactions.addLocalTransaction(transaction, validationResult.maybeAccount);
-
-      if (transactionAddedResult.isRejected()) {
-        final var rejectReason =
-            transactionAddedResult
-                .maybeInvalidReason()
-                .orElseGet(
-                    () -> {
-                      LOG.warn("Missing invalid reason for status {}", transactionAddedResult);
-                      return INTERNAL_ERROR;
-                    });
-        return ValidationResult.invalid(rejectReason);
-      }
-
+    final var result = addTransaction(transaction, true);
+    if (result.isValid()) {
+      localSenders.add(transaction.getSender());
```

Comment:
yes, it is the sender that gets the priority not the single tx, since it make no real sense to apply different priorities to txs sent by the same sender, and doing that could result in strange behavior, where a low priority tx could prevent following priority txs to be included.